### PR TITLE
D3ASIM-3307: Handled case for empty dict in the restore state methods.

### PR DESCRIPTION
**Pull request by spyrostz**

### CODE REVIEW

**Review by Hassan754 — APPROVED** [review]

File: d3a_interface/sim_results/kpi.py [hunk 1]
```diff
@@ -225,22 +225,24 @@ def update(self, area_dict, core_stats, current_market_slot):
...
+                last_known_state_data.get('total_energy_demanded_wh', 0.0)
             self.state[area_dict['name']].total_energy_produced_wh = \
-                last_known_state_data['total_energy_produced_wh']
+                last_known_state_data.get('total_energy_produced_wh', 0.0)
             self.state[area_dict['name']].total_self_consumption_wh = \
-                last_known_state_data['total_self_consumption_wh']
+                last_known_state_data.get('total_self_consumption_wh', 0.0)
             self.state[area_dict['name']].self_consumption_buffer_wh = \
-                last_known_state_data['self_consumption_buffer_wh']
+                last_known_state_data.get('self_consumption_buffer_wh', 0.0)
```
Comment (faizan2590) [inline d3a_interface/sim_results/kpi.py:245]: I would rather keep it strict as earlier, so at least we know via log to debug later why the targeted keys' value aren't there

**Review by faizan2590 — DISMISSED** [review]
One suggestion, other than that LGTM

**Review by hannesdiedrich — APPROVED** [review]
I agree with Faizan.
LGTM

File: d3a_interface/sim_results/kpi.py (hunk 1 above)
Comment (spyrostz) [reply d3a_interface/sim_results/kpi.py:245]: Added these in order to handle malformed messages too, but you are right, better to see the issue immediately. Reverting this

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**